packages/app-root/src/components/RootLayout.js

```diff
@@ -1,11 +1,16 @@
+import { cookies } from 'next/headers'
+
 import PageContextProviders from './PageContextProviders.js'
 import PageHeader from './PageHeader.js'
 import PageFooter from './PageFooter.js'
 
 export default function RootLayout({ children }) {
+
+  const storedThemeMode = cookies().get('theme')
+
   return (
     <body>
-      <PageContextProviders>
+      <PageContextProviders storedThemeMode={storedThemeMode?.value}>
```

Comment:
`storedThemeMode?.value` is undefined before the cookie is set, which leads to the bug I saw where the first click of the toggle does nothing.